operator/src/reference_values.rs

```diff
@@ -294,4 +298,78 @@ mod tests {
         let clos = |client| create_pcrs_config_map(client, Default::default());
         test_create_error(clos).await;
     }
+
+    fn dummy_job() -> Job {
+        Job {
+            metadata: ObjectMeta {
+                name: Some("test".to_string()),
+                ..Default::default()
+            },
+            status: Some(JobStatus {
+                completion_time: Some(Time(Utc::now())),
+                ..Default::default()
+            }),
+            ..Default::default()
+        }
+    }
+
+    fn pcr_response(req: &Request<Body>) -> Result<String, StatusCode> {
+        if req.method() == Method::DELETE {
+            Ok(serde_json::to_string(&Job::default()).unwrap())
+        } else if req.uri().path().contains(PCR_CONFIG_MAP) {
+            Ok(serde_json::to_string(&dummy_pcrs_map()).unwrap())
+        } else if req.uri().path().contains(trustee::TRUSTEE_DATA_MAP) {
+            Ok(serde_json::to_string(&ConfigMap {
+                data: Some(BTreeMap::from([(
+                    trustee::REFERENCE_VALUES_FILE.to_string(),
+                    "[]".to_string(),
+                )])),
+                ..Default::default()
+            })
+            .unwrap())
+        } else {
+            panic!("unexpected API interaction: {req:?}");
+        }
+    }
+
+    #[tokio::test]
+    async fn test_job_reconcile_success() {
+        let ctx = generate_rv_ctx(MockClient::new(pcr_response, "test".to_string()).into_client());
+        let job = Arc::new(dummy_job());
+        let result = job_reconcile(job, Arc::new(ctx)).await.unwrap();
+        assert_eq!(result, Action::await_change());
+    }
+
+    #[tokio::test]
+    async fn test_job_reconcile_no_name() {
+        let clos = |req: &Request<_>| panic!("unexpected API interaction: {req:?}");
+        let ctx = generate_rv_ctx(MockClient::new(clos, "test".to_string()).into_client());
+        let job = Arc::new(Default::default());
+        let err = job_reconcile(job, Arc::new(ctx)).await.err().unwrap();
+        assert!(err.to_string().contains("but had no name"));
+    }
+
+    #[tokio::test]
+    async fn test_job_reconcile_no_status() {
+        let clos = |req: &Request<_>| panic!("unexpected API interaction: {req:?}");
+        let ctx = generate_rv_ctx(MockClient::new(clos, "test".to_string()).into_client());
+        let mut job = dummy_job();
+        job.status = None;
+        let err = job_reconcile(Arc::new(job), Arc::new(ctx))
+            .await
+            .err()
+            .unwrap();
+        assert!(err.to_string().contains("but had no status"));
+    }
```

Comment:
I'm also not sure about this test as well, isn't kubernetes that set the job status?